### PR TITLE
Update version number to 2.3.0-rc1 and add release notes

### PR DESCRIPTION
**Pull request by EwoutH**
- Update version number to `2.3.0-rc1` 
- Add release notes

Please review and if good, approve, so I can tag the release.

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,6 +1,121 @@
...
 ---
+# 2.3.0-rc1 (2024-04-18)
+## Highlights
+The 2.3.0-rc1 release is our first release candidate pre-release, meant to test all the new features and enhancement for the upcoming 2.3.0 release.
+
+There are two main new features:
+- The experimental cell-centric discrete spaces, as added in #1994. It allows having cells with not only properties but also active behaviors: the `CellAgent`. Its inspired by NetLogo's [patches](https://ccl.northwestern.edu/netlogo/bind/primitive/patches.html) but extend and generalize this concept further.
+- Full support for discrete event scheduling, as added in #2066. It allows scheduling events (like Agent actions) at any time, including non-integer timesteps.
+
+There are a lot of other features: The Jupyter visualisation now supports sliders, `NetworkGrid.get_neighbors()` supports a radius, `AgentSet.get()` can retrieve multiple attributes and there are now benchmarks to track Mesa performance during development.
```
Comment (rht) [inline HISTORY.md:12]: The JupyterViz has always supported sliders, the PR was to add a concise convenient wrapper.

Comment (EwoutH) [reply HISTORY.md:12]: How would you formulate it exactly?

**Review by jackiekazil — APPROVED** [review]
I reviewed... I love the way releases go out now. It looks so much cleaner. Thank you.

File: HISTORY.md (hunk 1 above)
Comment (rht) [reply HISTORY.md:12]: @EwoutH this statement is misleading, and the PR shouldn't be merged before this is fixed.

Comment (EwoutH) [reply HISTORY.md:12]: Could you offer a suggestion?

Comment (rht) [reply HISTORY.md:12]: "The Jupyter visualisation now supports easier way to specify sliders" is short and not specific enough, but better than the previous one.

**Pull request merged**